lib/handsoap/service.rb

```diff
@@ -96,9 +96,10 @@ class Service
     @@create_document_callback = nil
     @@log_header_callback = nil
     @@log_body_callback = nil
-    attr_reader :http_client
+    attr_reader :http_client, :cookie
     def initialize(ep)
       @http_client = nil
+      @cookie = nil
```

Comment:
How about including the headers with the `Response` object so `login` could call `soap_response.cookie` for that specific response?